server/chat-commands/moderation.ts

```diff
@@ -814,39 +845,47 @@ exports.commands = {
 		}
 
 		const roomauth = Rooms.global.destroyPersonalRooms(userid);
-		if (roomauth.length) Monitor.log(`[CrisisMonitor] Globally banned user ${name} has public roomauth (${roomauth.join(', ')}), and should probably be demoted.`);
-
+		if (roomauth.length) {
+			Monitor.log(
+				`[CrisisMonitor] Globally banned user ${name} has public roomauth (${roomauth.join(', ')}), and should probably be demoted.`
+			);
+		}
 		let proof = '';
 		let userReason = target;
-		let targetLowercase = target.toLowerCase();
+		const targetLowercase = target.toLowerCase();
 		if (target && (targetLowercase.includes('spoiler:') || targetLowercase.includes('spoilers:'))) {
-			let proofIndex = (targetLowercase.includes('spoilers:') ? targetLowercase.indexOf('spoilers:') : targetLowercase.indexOf('spoiler:'));
-			let bump = (targetLowercase.includes('spoilers:') ? 9 : 8);
+			const proofIndex = (
+				targetLowercase.includes('spoilers:') ? targetLowercase.indexOf('spoilers:') :
+				targetLowercase.indexOf('spoiler:')
+			);
+			const bump = (targetLowercase.includes('spoilers:') ? 9 : 8);
 			proof = `(PROOF: ${target.substr(proofIndex + bump, target.length).trim()}) `;
 			userReason = target.substr(0, proofIndex).trim();
 		}
 
 		targetUser.popup(`|modal|${user.name} has globally banned you.${(userReason ? `\n\nReason: ${userReason}` : ``)} ${(Config.appealurl ? `\n\nIf you feel that your ban was unjustified, you can appeal:\n${Config.appealurl}` : ``)}\n\nYour ban will expire in a few days.`);
 
-		let banMessage = `${name} was globally banned by ${user.name}.${(userReason ? ` (${userReason})` : ``)}`;
+		const banMessage = `${name} was globally banned by ${user.name}.${(userReason ? ` (${userReason})` : ``)}`;
 		this.addModAction(banMessage);
 
 		// Notify staff room when a user is banned outside of it.
 		if (room.roomid !== 'staff' && Rooms.get('staff')) {
-			Rooms.get('staff').addByUser(user, `<<${room.roomid}>> ${banMessage}`);
+			Rooms.get('staff')?.addByUser(user, `<<${room.roomid}>> ${banMessage}`);
 		}
 
-		let affected = await Punishments.ban(targetUser, null, null, false, userReason);
-		let acAccount = (targetUser.autoconfirmed !== userid && targetUser.autoconfirmed);
+		const affected = await Punishments.ban(targetUser, null, null, false, userReason);
+		const acAccount = (targetUser.autoconfirmed !== userid && targetUser.autoconfirmed);
 		let displayMessage = '';
 		if (affected.length > 1) {
 			let guests = affected.length - 1;
-			affected = affected.slice(1).map(user => user.getLastName()).filter(alt => alt.substr(0, 7) !== '[Guest ');
+			(affected as any as string) = affected.slice(1)
+				.map((curUser: User) => curUser.getLastName())
+				.filter((alt: string) => alt.substr(0, 7) !== '[Guest ') as any as string;
```

Comment:
wtf?

`affected` isn't a string. If you don't know what's wrong here, just leave it as an error so I can see the error and fix it.

```suggestion
			affected = affected.slice(1)
				.map(curUser => curUser.getLastName())
				.filter(alt => !alt.startsWith('[Guest '));
```